core/vm/instructions.go

```diff
@@ -375,10 +375,15 @@ func opReturnDataCopy(pc *uint64, interpreter *EVMInterpreter, scope *ScopeConte
 
 func opExtCodeSize(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
 	slot := scope.Stack.Peek()
-	codeSize, err := interpreter.evm.IntraBlockState().ResolveCodeSize(slot.Bytes20())
+
+	codeSize, err := interpreter.evm.IntraBlockState().GetCodeSize(slot.Bytes20())
 	if err != nil {
 		return nil, fmt.Errorf("%w: %w", ErrIntraBlockStateFailed, err)
 	}
+	if codeSize == types.DelegateDesignationCodeSize {
```

Comment:
The size check is not enough. The prefix of the code should be checked as well (see `ParseDelegation`).

Reply:
Got it.